main.js

```diff
@@ -60,49 +61,44 @@ app.listen(port, () => {
     // (the ID is constant, the username can change at any time!)
     // msg.content has the message contents (e.g. text) as a string
     // msg.createdTimestamp has the message creation time (integer)
-
+    
     // Make sure DB is connected before we start inserting
-    msg.guild.members
+    roles = msg.guild.members
       .fetch(msg.author)
       .then((m) => {
         let roles = [];
         for (const role of m.roles.cache) {
           roles.push(role[1].name);
-        }
-        if (db_conn) {
-          console.log(
-            "writing to db",
-            [
-              Number(msg.createdTimestamp),
-              Number(msg.author.id),
-              msg.author.username,
-              msg.content,
-              msg.channel.name,
-              Number(msg.channelId),
-              roles,
-              msg.guild.name,
-            ].join(", ")
-          );
-          db.query(
-            "INSERT INTO discord (msg_time, author_id, author_name, msg, channel, channel_id, roles, server_name) VALUES ($1, $2, $3, $4, $5, $6, $7, $8)",
-            [
-              Number(msg.createdTimestamp),
-              Number(msg.author.id),
-              msg.author.username,
-              msg.content,
-              msg.channel.name,
-              Number(msg.channelId),
-              roles,
-              msg.guild.name,
-            ]
-          )
-            .then((result) => {})
-            .catch((e) => console.error(e.stack));
-        }
-      })
-      .then((result) => {})
-      .catch((e) => console.error(e.stack));
-  });
+        }})
+        .catch((e) => {
+          console.error("missing roles", e.stack)
+          return null})
+
+    console.log("messageCreate", msg.author.username, msg.content, msg.channel.name, msg.channelId, msg.guild.name, roles);
+    
+    row = [
+      Number(msg.createdTimestamp),
+      Number(msg.author.id),
+      msg.author.username,
+      msg.content,
+      msg.channel.name,
+      Number(msg.channelId),
+      roles,
+      msg.guild.name,
+    ]
+    if (db_conn) {
+      console.log(
+        "writing to db",
+        row.join(", ")
+      );
+      db.query(
+        "INSERT INTO discord (msg_time, author_id, author_name, msg, channel, channel_id, roles, server_name) VALUES ($1, $2, $3, $4, $5, $6, $7, $8)",
+        row
+      )
+        .then((result) => {})
+        .catch((e) => console.error(e.stack));
```

Comment:
could we add a log for successful write as well?

Reply:
Done!